### PR TITLE
Add ClaimRewardTx for deposit

### PR DESCRIPTION
**Pull request by evlekht**
This PR adds claimRewardTx, which mint reward outputs for specified deposit transactions.

ClaimRewardTx must have additional credential for deposit transactions. This credential must contain signatures for all deposit transaction reward owners, order doesn't matter.

Reward outputs will be owned by claimRewardTx.RewardOwner, unless its empty. If it is empty, reward output will be owned by depositTx.RewardOwner.

### CODE REVIEW

File: vms/platformvm/txs/builder/camino_builder.go [hunk 1]
```diff
@@ -59,6 +63,13 @@ type CaminoTxBuilder interface {
 		change *secp256k1fx.OutputOwners,
 	) (*txs.Tx, error)
 
+	NewClaimRewardTx(
+		depositTxIDs []ids.ID,
+		rewardAddress ids.ShortID,
```
Comment (c4t-ag) [inline vms/platformvm/txs/builder/camino_builder.go:68]: We have pimped everything to OutputOwners, should contine here, too
If you change it, pls. don't forget the msig verification call.

File: vms/platformvm/txs/builder/camino_helpers_test.go [hunk 2]
```diff
@@ -491,3 +525,22 @@ func generateTestUTXO(txID ids.ID, assetID ids.ID, amount uint64, outputOwners s
 	testUTXO.InputID()
 	return testUTXO
 }
+
+func generateKeyAndOwner() (*crypto.PrivateKeySECP256K1R, ids.ShortID, secp256k1fx.OutputOwners) {
+	key, err := testKeyfactory.NewPrivateKey()
```
Comment (c4t-ag) [inline vms/platformvm/txs/builder/camino_helpers_test.go:530]: Just want to mention: There is a crypto.generatetestKeys, too

File: vms/platformvm/txs/builder/camino_builder.go [hunk 3]
```diff
@@ -347,6 +358,59 @@ func (b *caminoBuilder) NewUnlockDepositTx(
...
+
+	utx := &txs.ClaimRewardTx{
+		BaseTx: txs.BaseTx{BaseTx: avax.BaseTx{
+			NetworkID:    b.ctx.NetworkID,
+			BlockchainID: b.ctx.ChainID,
+			Ins:          ins,
+			Outs:         outs,
+		}},
+		DepositTxs: depositTxIDs,
+		RewardsOwner: &secp256k1fx.OutputOwners{
```
Comment (c4t-ag) [inline vms/platformvm/txs/builder/camino_builder.go:401]: See above ^^ we should maybedirectly to allow OutputOwners from api

File: vms/platformvm/txs/codec.go [hunk 4]
```diff
@@ -112,6 +112,7 @@ func RegisterUnsignedTxsTypes(targetCodec codec.CaminoRegistry) error {
 		targetCodec.RegisterCustomType(&AddressStateTx{}),
 		targetCodec.RegisterCustomType(&DepositTx{}),
 		targetCodec.RegisterCustomType(&UnlockDepositTx{}),
+		targetCodec.RegisterCustomType(&ClaimRewardTx{}),
```
Comment (c4t-ag) [inline vms/platformvm/txs/codec.go]: Can we please add it at the end? ID's are already in magellan and caminojs

File: vms/platformvm/txs/executor/camino_tx_executor.go [hunk 5]
```diff
@@ -639,6 +645,118 @@ func (e *CaminoStandardTxExecutor) UnlockDepositTx(tx *txs.UnlockDepositTx) erro
...
+		}
+
+		// checking deposit signatures
+
+		depositRewardsOwner, ok := depositTx.RewardsOwner.(*secp256k1fx.OutputOwners)
+		if !ok {
+			return errNotSECPOwner
+		}
+
+		if err := e.Fx.VerifyPermissionUnordered(tx, e.Tx.Creds[len(e.Tx.Creds)-1], depositRewardsOwner); err != nil {
```
Comment (c4t-ag) [inline vms/platformvm/txs/executor/camino_tx_executor.go:704]: This doesn't look like to hate to be inside loop

File: vms/platformvm/txs/executor/camino_tx_executor.go [hunk 6]
```diff
@@ -639,6 +645,118 @@ func (e *CaminoStandardTxExecutor) UnlockDepositTx(tx *txs.UnlockDepositTx) erro
...
+
+		depositOffer, err := e.State.GetDepositOffer(deposit.DepositOfferID)
+		if err != nil {
+			return err
+		}
+
+		claimableReward := deposit.ClaimableReward(depositOffer, currentTimestamp)
+		if claimableReward > 0 {
+			rewardsOwner := tx.RewardsOwner
+			secpOwners, ok := rewardsOwner.(*secp256k1fx.OutputOwners)
```
Comment (c4t-ag) [inline vms/platformvm/txs/executor/camino_tx_executor.go:723]: This block also outside loop?

**Review by c4t-ag — CHANGES_REQUESTED** [review]

File: vms/platformvm/txs/builder/camino_builder.go (hunk 1 above)
Comment (evlekht) [reply vms/platformvm/txs/builder/camino_builder.go:68]: I don't think that we changed camino tx builder methods - they all (validator, deposit,..) still accept rewardAddress and then create 1-addr owner
Should I also change them?

File: vms/platformvm/txs/executor/camino_tx_executor.go (hunk 5 above)
Comment (evlekht) [reply vms/platformvm/txs/executor/camino_tx_executor.go:704]: It uses depositRewardsOwner, which is `depositRewardsOwner, ok := depositTx.RewardsOwner.(*secp256k1fx.OutputOwners)`, and depositTx is from `for _, depositTxID := range tx.DepositTxs {`

File: vms/platformvm/txs/executor/camino_tx_executor.go (hunk 6 above)
Comment (evlekht) [reply vms/platformvm/txs/executor/camino_tx_executor.go:723]: same as previous comment - its reward for 1 specific deposit, which is loop iteration

**Review by c4t-ag — APPROVED** [review]

**Pull request merged**